benchmark/bull/harness.js

```diff
@@ -1,26 +1,50 @@
-var Queue = require('bull');
-var queue = new Queue('test');
+const helpers = require('../../lib/helpers');
+const Queue = require('bull');
+const queue = new Queue('test');
 
-module.exports = function (options, cb) {
-  var finished = 0;
-  var finishTime, startTime;
-
-  var reportResult = function (result) {
-    finished += 1;
-    if (finished === options.numRuns) {
-      finishTime = (new Date()).getTime();
-      cb(null, finishTime - startTime);
+// A promise-based barrier.
+function reef(n = 1) {
+  const done = helpers.deferred(), end = done.defer();
+  return {
+    done,
+    next() {
+      --n;
+      if (n < 0) return false;
+      if (n === 0) end();
+      return true;
     }
   };
+}
 
-  queue.process(options.concurrency, function (job, done) {
-    reportResult();
-    return done();
-  });
+module.exports = (options) => {
```

Comment:
Can we wrap the body of this in a `queue.isReady().then(...)`? I believe that would address the concern here about startup costs/Lua script caching time: https://github.com/bee-queue/bee-queue/issues/67#issuecomment-322008724

Reply:
Fixed